note: 'value' has been explicitly marked unavailable here

### Platform

libretro / Retroarch

### Compiler and build tool versions

AppleClang 12.0.0.12000032

### Operating system version

macOS

### Build commands used

I'm afraid I don't own a mac, but this is off the libretro build bot:

```
[ 70%] Building CXX object CMakeFiles/Core.dir/Core/CmdLine.cpp.o
warning: unknown warning option '-Wno-unused-but-set-variable'; did you mean '-Wno-unused-const-variable'? [-Wunknown-warning-option]
warning: unknown warning option '-Wno-deprecated-builtins'; did you mean '-Wno-deprecated-volatile'? [-Wunknown-warning-option]
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:369:37: error: 'value' is unavailable: introduced in macOS 10.14
                g_Config.bFullScreen = fullscreen.value();
                                                  ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:373:42: error: 'value' is unavailable: introduced in macOS 10.14
                g_Config.iGPUBackend = (int)gpuBackend.value();
                                                       ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:377:51: error: 'value' is unavailable: introduced in macOS 10.14
                g_Config.bSoftwareRendering = softwareRendering.value();
                                                                ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:381:43: error: 'value' is unavailable: introduced in macOS 10.14
                g_Config.bEnableLogging = enableLogging.value();
                                                        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:389:36: error: 'value' is unavailable: introduced in macOS 10.14
                g_Config.iCpuCore = (int)cpuCore.value();
                                                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:392:45: error: 'value' is unavailable: introduced in macOS 10.14
                g_Config.bPauseExitsEmulator = escapeExit.value();
                                                          ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:395:52: error: 'value' is unavailable: introduced in macOS 10.14
                g_Config.bPauseMenuExitsEmulator = pauseMenuExit.value();
                                                                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:399:41: error: 'value' is unavailable: introduced in macOS 10.14
                g_logManager.SetAllLogLevels(logLevel.value());
                                                      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:403:36: error: 'value' is unavailable: introduced in macOS 10.14
                g_Config.bUseOldAtrac = oldAtrac.value();
                                                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:409:34: error: 'value' is unavailable: introduced in macOS 10.14
                g_Config.mountRoot = Path(root.value());
                                               ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
/Users/gitlab/builds/FkGsyN3zF/0/libretro/ppsspp/Core/CmdLine.cpp:413:50: error: 'value' is unavailable: introduced in macOS 10.14
                g_Config.iInternalResolution = resolutionScale.value();
                                                               ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/optional:944:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
2 warnings and 11 errors generated.
make[3]: *** [CMakeFiles/Core.dir/Core/CmdLine.cpp.o] Error 1
make[2]: *** [CMakeFiles/Core.dir/all] Error 2
make[1]: *** [libretro/CMakeFiles/ppsspp_libretro.dir/rule] Error 2
```

### What happens

Building failing on the libretro buildbot at the moment: https://git.libretro.com/libretro/ppsspp/-/jobs/1898908/raw


### PPSSPP version affected

master

### Last working version

_No response_

### Checklist

- [x] Test in the [latest git build](https://www.ppsspp.org/devbuilds) in case it's already fixed.
- [x] Make sure to run `git submodule update --init --recursive` before building.
- [x] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [x] Try deleting the build directory and running the build again.
- [x] Check GitHub Actions, which builds every merge and PR.
- [x] Include logs and help us reproduce.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2137,13 +2137,21 @@ if((MACOSX OR IOS) AND NOT LIBRETRO)
 		Core/Util/DarwinFileSystemServices.h)
 endif()
 
+set(CmdLine)
+
+if(NOT LIBRETRO)
+    list(APPEND CmdLine
+        Core/CmdLine.cpp
+        Core/CmdLine.h
+    )
+endif()
+
 # 'ppsspp_jni' on ANDROID, 'Core' everywhere else
 # SHARED on ANDROID, STATIC everywhere else
 add_library(${CoreLibName} ${CoreLinkType}
 	${CoreExtra}
 	${CommonJIT}
-	Core/CmdLine.cpp
-	Core/CmdLine.h
+	${CmdLine}
 	Core/Config.cpp
 	Core/Config.h
 	Core/ConfigSettings.cpp

